src/app/users/users.component.ts

```diff
@@ -0,0 +1,83 @@
+import { Component, OnInit } from '@angular/core';
+import { UsersObject } from '../../../api-objects/UsersObject';
+// import * as createAccount from '../services/createAccount'; // Uncomment when available
+
+@Component({
+  selector: 'users',
+  templateUrl: './users.component.html',
+  styleUrls: ['./users.component.css']
+})
+
+export class UsersComponent implements OnInit {
+
+  public username: string = '';
+  public userPassword: string = 'htxliwq7ja'; // example temp password to be displayed to the user(Remove later)
+  public authorityType: string = '';
+  public userEmail: string = '';
+  public name: string = '';
+  public usernameTaken: boolean = true;
+  public showUserInfo: boolean = false;
+
+  constructor() {}
+
+  public ngOnInit() {
+  }
+
+/** Checks whether a username exists or not, sets the temp password if the user doesn't exist, 
+ *  and returns true or false based on if the user exists or not.
+ *  @Param username: Checked for uniqueness by the createAccount call
+ */
+  public checkUserName(username:string): boolean {
+  	/*try{
+      let userAvailable = createAccount(username);
```

Comment:
If this is where you want to talk to the server then all the fields should be validated first.  This should then also take a UsersObject instead of a string.

Reply:
Agreed, I was admittedly mistaken beforehand: the API will catch if a username already exists. So there isn't a need for a checkUserName component

Reply:
Yeah checkUserName, was meant to call the server, but since the service wasn't implemented, I made it a simple check to see if the username field is empty or not. However, I realized that I do that check in createUser as well. So, checkUserName servers no purpose right now, but it's better to keep and refactor it once the service is done to split up the work, instead of making createUser do everything.